src/test/java/mcscheduler/logic/commands/TakeLeaveCommandTest.java

```diff
@@ -170,6 +171,11 @@ public ModelStubAlreadyHasAssignment(List<Worker> workers, List<Shift> shifts) {
         public boolean hasAssignment(Assignment assignment) {
             return true;
         }
+
+        @Override
+        public Optional<Assignment> getAssignment(Assignment toGet) {
+            return Optional.of(toGet);
+        }
```

Comment:
Don't think this is necessary as `getAssignment` was added to this stub class in a recent PR. But perhaps that can be updated to either fetch or always provide `toGet` using `orElse()` to fulfil this functionality.

Reply:
alright, fixed!